### PR TITLE
Set up pattern library

### PR DESCRIPTION
**Pull request by victoriachan**
# Description

Link to sample test page: http://localhost:8000/pattern-library/
Related PRs/issues: #9665 

To enable pattern-library locally, add `PATTERN_LIBRARY_ENABLED=1` to .env file. Then visit the pattern library page at http://localhost:8000/pattern-library/ 

IMPT: Pattern Library should not be enabled on production sites. Eventually we can add `PATTERN_LIBRARY_ENABLED=1` to docker-compose so it is enabled on local docker containers by default. 

I have not moved the main templates from `network-api/networkapi/templates`, as it seems to work without having to move it. 

It will take time to adapt all the templates to render well in Pattern Library. This will involve creating override tags and dummy context in the `.yaml` files for each template. The changes should not affect the normal workings of the site. 

# Done in this PR:

- I have also added a new app: `networkapi.project_styleguide` to house the [tag overrides](https://torchbox.github.io/django-pattern-library/guides/overriding-template-tags/) that are needed for rendering the templates in pattern libary.
- I have created a new `pattern_library_base.html` template based on `pages/base.html`. Pattern library requires a base template for wrapping the non-page templates. There are duplicate content between this and `pages/base.html` as I didn't want to affect the existing structure too much. Eventually we could update `pages/base.html` to extend from a base template and use that for pattern library instead. Currently `pattern_library_base.html` is only used by pattern library.
- Added template context for article_page.html (though the body streamfield blocks still needs adding). You can view this at locally http://localhost:8000/pattern-library/pattern/pages/article_page.html You can also view the fragments separately at, e.g. http://localhost:8000/pattern-library/pattern/fragments/publication_hero.html

# Issues:

- I don't know why but the left hand navigation on pattern library is displaying duplicate links for each section for me. 

# TODO (next phase?):

- Update `pages/base.html` to extend from `pattern_library_base.html`.
- Add `PATTERN_LIBRARY_ENABLED=True` to docker-compose so it is enabled on local docker containers by default.
- Add documentation to README etc. 
- Checklist to PR templates to ensure template context is maintained.

# Checklist

**Tests**
- [ ] Is the code I'm adding covered by tests?

**Changes in Models:**
- [x] Did I update or add new fake data?
- [x] Did I squash my migration?
- [x] Are my changes [backward-compatible](https://github.com/mozilla/foundation.mozilla.org/blob/main/docs/workflow.md#django-migrations-what-to-do-when-working-on-backward-incompatible-migrations). If not, did I schedule a deploy with the rest of the team?

**Documentation:**
- [ ] Is my code documented?
- [ ] Did I update the READMEs or wagtail documentation?

### CODE REVIEW

**Review by jhonatan-lopes — CHANGES_REQUESTED** [review]
Thanks @victoriachan! Looking good and working well locally. I just have some minor comments.

**Comment by jhonatan-lopes** [comment]
Hi @victoriachan, another thing that I remembered is that we also need to enable the `CSP_FRAME_ANCESTORS` to make the pattern library work (see #10768). Could you update the documentation with this information as well, please?

File: network-api/networkapi/project_styleguide/templatetags/localization.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+from pattern_library.monkey_utils import override_tag
+
+from networkapi.wagtailpages.templatetags.localization import register
+
+override_tag(register, name="get_unlocalized_url", default_html="/dummy")
+override_tag(register, name="relocalized_url", default_html="/en/dummy")
+override_tag(register, name="localizedroutablepageurl", default_html="/en/routed-dummy")
```
[the remark this answers is not in the document]
Comment (victoriachan) [reply network-api/networkapi/project_styleguide/templatetags/localization.py:14]: Thanks, @jhonatan-lopes. I have added more comments to that now.

File: network-api/networkapi/settings.py [hunk 2]
```diff
@@ -261,6 +262,9 @@
             "networkapi.wagtailpages",
             "networkapi.mozfest",
             "networkapi.donate",
+            # TODO: only enable on dev
+            "pattern_library",
```
[the remark this answers is not in the document]
Comment (victoriachan) [reply network-api/networkapi/settings.py]: Ah thanks. I have updated this now.

File: network-api/networkapi/settings.py [hunk 3]
```diff
@@ -722,3 +730,36 @@
...
+# https://torchbox.github.io/django-pattern-library/
+#
+# Pattern library isn’t intended for production usage, and hasn’t received
+# extensive security scrutiny. Don't enable it on production.
+#
+# PATTERN_LIBRARY_ENABLED is set to True in docker-compose.yml for local development.
+# For pattern library to work with CSP, you also need to add the following to your .env file:
+#    PATTERN_LIBRARY_ENABLED=1
+#    X_FRAME_OPTIONS=SAMEORIGIN
+#    CSP_FRAME_ANCESTORS="'self'"
```
Comment (victoriachan) [inline network-api/networkapi/settings.py:744]: @jhonatan-lopes I see your comments about the CSP fixes. I have added the instructions on the settings file now. Is this where you have in mind to add the documentation?

Comment (jhonatan-lopes) [reply network-api/networkapi/settings.py:744]: This is perfect, thanks!

**Review by jhonatan-lopes — APPROVED** [review]
Thank you for the changes, @victoriachan! LGTM :+1:

**Comment by mmmavis** [comment]
Same issue as last time - Percy doesn't run on non-Moz branch. I'm gonna bypass the check and merge this PR.

**Comment by mmmavis** [comment]
Reverting as it's producing errors

```
django.template.library.InvalidTemplateLibrary: Invalid template library specified. ImportError raised when trying to load 'networkapi.project_styleguide.templatetags.localization': No module named 'pattern_library'
```

**Pull request merged**